private/buf/cmd/buf/command/push/push.go

```diff
@@ -183,6 +184,9 @@ func run(
 		}
 		return err
 	}
+	if localModulePin == nil {
+		return errors.New("Missing local module pin in the registry's response.")
+	}
```

Comment:
Don't see a scenario in which `service.Push` does `return nil, nil`, also in theory that [shouldn't happen](https://bufprivate.slack.com/archives/C02KXVDB23B/p1665590362168519?thread_ts=1665589006.759369&cid=C02KXVDB23B)? not very idiomatic.

Reply:
`service.Push` isn't Go. Well, it is but only on the surface.

It's code generated from `proto/buf/alpha/registry/v1alpha1/push.proto` and return values are whatever the remote implementation gives you, if anything, because `PushResponse.LocalModulePin` is optional.

Reply:
This one is a bit of an interesting one for me, I think defensively, our clients should handle that case. In practice, it should never be empty because a push should not be able to succeed without being able to respond with what the new commit is, and thus should be handled on the server-side (which is what we do now). That being said, this error would indicate to us if the server-side no longer handles that case, so I'm in favour of this.

Reply:
Exactly. It's a pretty silly response since a pin is needed, but only because that's our understanding today. Perhaps a new field replaces it and this old client won't know what to do. :)